File: budget/src/main/java/com/taubel/budget/entities/LineItem.java
package com.taubel.budget.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "Line_Items")
public class LineItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "line_item_id")
    private Long id;
    
    private String name;

    @Column(name = "budgeted_amount")
    private double budgetedAmount;

    @ManyToOne()
    @JoinColumn(name = "category_id")
    private Category category;

    @ManyToOne()
    @JoinColumn(name = "user_id")
    private User user;  

}

File: budget-frontend/src/app/features/budget/data/models/lineitem.ts
export class LineItem {
    id: number
    name: string
    budgetedAmount: number
    userId: number
    categoryId: number

  constructor(
    id: number, 
    name: string, 
    budgetedAmount: number, 
    userId: number, 
    categoryId: number
) {
    this.id = id
    this.name = name
    this.budgetedAmount = budgetedAmount
    this.userId = userId
    this.categoryId = categoryId
  }    
}